### PR TITLE
Fix import export file directory

### LINKED ISSUE #305: [PE-D2] Import not working
Screenshot 2021-11-05 at 7 22 52 PM

Import not working

### PR DESCRIPTION
**Pull request by kevinmingtarja**
Closes #305 

Desc: Now, the way I specify the directory to save it is by getting the `current directory` where the JAR file lives, then into the data directory. Previously, it depends on the `working directory` which might not be the same as the directory where the JAR file is located, hence due to this discrepancy, the bug that @hsiaotingluv encountered happened (since I assume you double clicked it instead of using `java -jar`). The previous implementation only works when you do `java -jar` from the directory of the JAR file, which is really prone to user error.

In short, my fix makes it less prone to error since I no longer hard code the path. For safety, can anyone with windows check if it works by building the JAR file from this branch and test import export.

WARNING: Due to this fix, the directory used to import/export when working in your local environment is also inevitably changed. But I think this is okay since the priority should be for it to work with our JAR file.

https://user-images.githubusercontent.com/69668484/140567541-99feb77c-368b-4d76-956f-0e304c933a6d.mov

### CODE REVIEW

File: src/main/java/seedu/address/commons/util/FileUtil.java [hunk 1]
```diff
@@ -80,4 +88,19 @@ public static void writeToFile(Path file, String content) throws IOException {
...
+    public static String getCurrentPath() throws URISyntaxException {
+        return String.valueOf(Main.class.getProtectionDomain().getCodeSource().getLocation().toURI().getPath());
+    }
+
+    /**
+     * Returns true if a given string is a valid file name.
+     */
+    public static boolean isValidJsonFileName(String test) {
+        if (test.isEmpty()) {
+            return true;
```
Comment (choonyongchan) [inline src/main/java/seedu/address/commons/util/FileUtil.java:100]: Just to clarify, should this return False? Should a empty string be considered a valid json file name

File: src/main/java/seedu/address/logic/commands/ImportCommand.java [hunk 2]
```diff
@@ -31,40 +33,44 @@
             + "Parameters: FILENAME\n"
             + "Example: " + COMMAND_WORD + " t35.json";
 
-    public static final String MESSAGE_SUCCESS = "File successfully imported. %1$d new contacts added";
+    public static final String MESSAGE_SUCCESS = "File successfully imported from.\n%s new contacts added";
```
Comment (choonyongchan) [inline src/main/java/seedu/address/logic/commands/ImportCommand.java:36]: Perhaps can remove 'from'. Since the sentence will ends at 'File successfully imported from.'

File: src/main/java/seedu/address/logic/parser/ExportCommandParser.java [hunk 3]
```diff
@@ -16,7 +18,7 @@
     public ExportCommand parse(String args) throws ParseException {
         String fileName = args.trim();
 
-        if (!fileName.endsWith(".json") || fileName.equals(".json")) {
+        if (!fileName.endsWith(".json") || fileName.equals(".json") || !isValidJsonFileName(fileName)) {
```
Comment (choonyongchan) [inline src/main/java/seedu/address/logic/parser/ExportCommandParser.java:21]: Not a bug. Having && is easier to follow than ||. Perhaps we can use De Morgans to convert || to &&

File: src/main/java/seedu/address/logic/parser/ImportCommandParser.java [hunk 4]
```diff
@@ -16,7 +18,7 @@
     public ImportCommand parse(String args) throws ParseException {
         String fileName = args.trim();
 
-        if (!fileName.endsWith(".json") || fileName.equals(".json")) {
+        if (!fileName.endsWith(".json") || fileName.equals(".json") || !isValidJsonFileName(fileName)) {
```
Comment (choonyongchan) [inline src/main/java/seedu/address/logic/parser/ImportCommandParser.java:21]: ^ previous comment. Not a bug

**Review by choonyongchan — COMMENTED** [review]
Great job!! Some minor nits to glance through and it'll be perfect :)

File: src/main/java/seedu/address/commons/util/FileUtil.java (hunk 1 above)
Comment (kevinmingtarja) [reply src/main/java/seedu/address/commons/util/FileUtil.java:100]: Oh yeah right thanks for pointing it out

File: src/main/java/seedu/address/logic/parser/ExportCommandParser.java (hunk 3 above)
Comment (kevinmingtarja) [reply src/main/java/seedu/address/logic/parser/ExportCommandParser.java:21]: I think this one is still quite understandable

**Review by choonyongchan — APPROVED** [review]
Great work!
